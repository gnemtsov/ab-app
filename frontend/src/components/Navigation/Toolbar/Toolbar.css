.Toolbar {
    width: 100%;
    background-color: #1d2583;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 1.2rem;
}

.Logo {
    margin: 0.6rem;
    width: 8rem;
}

.DesktopOnly {
    display: none;
}


@media (min-width: 499px) {
    .DesktopOnly {
        display: block;
    }
}