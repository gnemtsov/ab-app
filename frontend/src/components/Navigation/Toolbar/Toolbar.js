import React from 'react';

import classes from './Toolbar.css';
import Logo from '../../Logo/Logo';
import NavigationItems from '../NavigationItems/NavigationItems';
import DrawerToggle from '../SideDrawer/DrawerToggle/DrawerToggle';

const toolbar = ( props ) => (
    <header className={classes.Toolbar}>
        <DrawerToggle clicked={props.drawerToggleClicked} />
        <div className={classes.Logo}>
            <Logo />
        </div>
        <nav className={classes.DesktopOnly}>
            <NavigationItems isAuthenticated={props.isAuth} />
        </nav>
    </header>
);

export default toolbar;