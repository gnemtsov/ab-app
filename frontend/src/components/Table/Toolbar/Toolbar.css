.Container {
    position: relative;
}

.Toolbar {
    position: relative;
    visibility: hidden;
    display: grid;
    grid-template-columns: min-content;
    grid-row-gap: 0.1rem;
    align-items: center;
}

@media (min-width: 40rem) {
    .Toolbar {
    }
}

.Toolbar.Show {
    visibility: visible;
}

.Toolbar button {
    border-radius: 0 6px 6px 0;
    padding: 0.2rem 0.4rem;
}