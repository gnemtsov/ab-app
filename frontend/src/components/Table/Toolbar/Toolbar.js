import React, { Component } from 'react';

import classes from './Toolbar.css';
import Button from '../../UI/Button/Button';

export default class Toolbar extends Component {

    conf = { 
        toolbarOffset: 30
    }

    state = {
        windowScrollY: 0,
        toolbarHeight: 0
    }

    constructor(props) {
        super(props);
        this.refToolbar = React.createRef();
        this.state.windowScrollY = window.scrollY;
    }

    componentDidMount() {
        window.addEventListener('scroll', this.updateScrollY);
        this.setState({ toolbarHeight: this.refToolbar.current.offsetHeight });
    }

    componentWillUnmount() {
        window.removeEventListener('scroll', this.updateScrollY);
    }

    updateScrollY = () => this.setState({ windowScrollY: window.scrollY })

    render() {
        let buttons = [];

        const {
            toolbarOffset
        } = this.conf;

        const {
            windowScrollY,
            toolbarHeight
        } = this.state;

        const {
            show,
            defaultTop,
            boundaryTop,
            boundaryBottom,
            defaultSortHandler,
            csvExportHandler,
            selectAllHandler
        } = this.props;

        let toolbarTop = defaultTop;
        if (windowScrollY > boundaryTop - toolbarOffset  && windowScrollY < boundaryBottom - toolbarHeight - toolbarOffset * 2) {
            toolbarTop += windowScrollY - boundaryTop + toolbarOffset;
        }

        if (typeof defaultSortHandler === 'function') {
            buttons.push(
                <Button
                    key="defaultSort"
                    clicked={defaultSortHandler}>
                    {defaultSortHandler.icon || '\u21C5'}
                </Button>
            );
        }

        if (typeof csvExportHandler === 'function') {
            buttons.push(
                <Button
                    key="csvExport"
                    clicked={csvExportHandler}>
                    {csvExportHandler.icon || 'csv'}
                </Button>
            );
        }

        if (typeof selectAllHandler === 'function') {
            buttons.push(
                <Button
                    key="selectAll"
                    clicked={selectAllHandler}>
                    {selectAllHandler.icon || '\u273D'}
                </Button>
            );
        }

        return (
            <div
                className={classes.Container}>
                <div
                    ref={this.refToolbar}
                    className={classes.Toolbar + (show ? ' ' + classes.Show : '')}
                    style={{ top: toolbarTop + 'px' }}>
                    {buttons}
                </div>
            </div>
        );
    }
}
