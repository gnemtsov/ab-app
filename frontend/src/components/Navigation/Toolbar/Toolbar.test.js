import React from 'react';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Toolbar from './Toolbar';

configure({ adapter: new Adapter() });

describe('Toolbar', () => {
	it('Render not authenticated', () => {
		const wrapper = shallow (
	        <Toolbar 
				isAuth={false}
				/>
		);
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Render authenticated', () => {
		const wrapper = shallow (
	        <Toolbar 
				isAuth={true}
				/>
		);
		
		expect(wrapper).toMatchSnapshot();
	});
});
