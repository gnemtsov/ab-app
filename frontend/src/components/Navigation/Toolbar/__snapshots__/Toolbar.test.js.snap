// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Render authenticated 1`] = `
<header
  className="Toolbar"
>
  <drawerToggle />
  <Link
    replace={false}
    to="/"
  >
    <img
      alt="AB-APP logo"
      className="Logo"
      src="test-pic-stub"
    />
  </Link>
  <nav
    className="DesktopOnly"
  >
    <navigationItems
      isAuthenticated={true}
    />
  </nav>
</header>
`;

exports[`Toolbar Render not authenticated 1`] = `
<header
  className="Toolbar"
>
  <drawerToggle />
  <Link
    replace={false}
    to="/"
  >
    <img
      alt="AB-APP logo"
      className="Logo"
      src="test-pic-stub"
    />
  </Link>
  <nav
    className="DesktopOnly"
  >
    <navigationItems
      isAuthenticated={false}
    />
  </nav>
</header>
`;
